cmd/ipfs/daemon.go

```diff
@@ -488,3 +503,16 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func runGC(req cmds.Request, repo repo.Repo, node *core.IpfsNode) error {
+	disableGC, _, err := req.Option(disableGCKwd).Bool()
+	if err != nil {
+		return err
+	}
+	if disableGC {
+		return nil
+	}
+
+	ctx := req.Context()
+	return fsrepo.PeriodicGC(ctx, repo, node)
```

Comment:
Moved to `corerepo`. Because `GarbageCollect` is a `corerepo`-tier code that any code that depends on it shouldn't be put in a lower level `repo`. Additionally, `PeriodicGC`'s code is generic that it doesn't depend on unixfs/s3-specific functions.